js/data.js

```diff
@@ -6,3 +6,39 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+
+var $imageURL = document.querySelector('input[name="imageUpload"]');
+var $textArea = document.querySelector('textarea[name="notes"]');
+var $title = document.querySelector('input[name="title"]');
+var $img = document.querySelector('img');
+
+$imageURL.addEventListener('input', function (event) {
+  $img.setAttribute('src', $imageURL.value);
+
+});
+
+var $entryForm = document.querySelector('div.entry-form');
+$entryForm.addEventListener('submit', function (event) {
+  var inputVals = {};
+  inputVals.imageURL = $imageURL.value;
+  inputVals.title = $title.value;
+  inputVals.textarea = $textArea.value;
+  inputVals.entryNumber = data.nextEntryId;
```

Comment:
make this `entryId`, not `entryNumber`. `id` is not for counting, its for a unique label